src/nrepl/jack-in-dependency-versions.ts

```diff
@@ -0,0 +1,225 @@
+  'cider/piggieback': 'cider/piggieback',
+};
+
+const JACK_IN_DEPENDENCY_KEYS = Object.keys(
+  JACK_IN_DEPENDENCY_LIBRARIES
+) as JackInDependencyKey[];
+
+const JACK_IN_DEPENDENCY_DEFAULTS: Record<JackInDependencyKey, string> = {
+  'nrepl': '1.3.1',
+  'cider-nrepl': '0.55.4',
+  'cider/piggieback': '0.6.0',
+};
+
+const GLOBAL_STATE_KEY = 'calva.jackIn.latestDependencyVersions';
+
+let refreshPromise: Promise<void> | null = null;
+
+function execFileAsync(command: string, args: string[]) {
+  return new Promise<{ stdout: string; stderr: string }>((resolve, reject) => {
+    child.execFile(
+      command,
+      args,
+      { encoding: 'utf8', timeout: 30000, maxBuffer: 1024 * 1024, windowsHide: true },
+      (error, stdout, stderr) => {
+        if (error) {
+          const execError = error as NodeJS.ErrnoException & {
+            stdout?: string;
+            stderr?: string;
+          };
+          execError.stdout = stdout;
+          execError.stderr = stderr;
+          reject(execError);
+          return;
+        }
+        resolve({ stdout, stderr });
+      }
+    );
+  });
+}
+
+function parseFindVersionsOutput(output: string): string | undefined {
+  return output
+    .split(/\r?\n/)
+    .map((line) => line.trim())
+    .filter((line) => line.length > 0)
+    .map((line) => {
+      try {
+        return parseEdn(line);
+      } catch (error) {
+        console.warn('[Calva] Failed to parse find-versions output line', line, error);
+        return undefined;
+      }
+    })
+    .map((data) => (data ? data['mvn/version'] : undefined))
+    .find((version) => typeof version === 'string');
+}
+
+function getDepsCljJarPath(): string | undefined {
+  const context = state.extensionContext;
+  if (!context) {
+    return undefined;
+  }
+  const jarPath = path.join(context.extensionPath, 'deps.clj.jar');
+  return fs.existsSync(jarPath) ? jarPath : undefined;
+}
+
+async function fetchLatestVersion(library: string): Promise<string> {
+  const args = ['-X:deps', 'find-versions', ':lib', library, ':n', '1'];
+  const errors: string[] = [];
+
+  try {
+    const { stdout } = await execFileAsync('clojure', args);
+    const version = parseFindVersionsOutput(stdout);
+    if (version) {
+      return version;
+    }
+    errors.push(`clojure output missing version for ${library}`);
+  } catch (error) {
+    errors.push(`clojure failed: ${(error as Error).message}`);
+  }
+
+  const depsCljJarPath = getDepsCljJarPath();
+  if (depsCljJarPath) {
+    try {
+      const { stdout } = await execFileAsync('java', ['-jar', depsCljJarPath, ...args]);
+      const version = parseFindVersionsOutput(stdout);
+      if (version) {
+        return version;
+      }
+      errors.push(`deps.clj output missing version for ${library}`);
+    } catch (error) {
+      errors.push(`deps.clj failed: ${(error as Error).message}`);
+    }
+  } else {
+    errors.push('deps.clj.jar not available');
+  }
+
+  throw new Error(errors.join(' | '));
+}
```

Comment:
Since Calva already download deps.clj jar, added this fallback to try to get the versions if the first approach fails.
I don't know if is needed to be honest, adds more complexity and I can't argue if when it fails by any reason using clojure cli, if this will be a good fallback.

Reply:
I think this is great. So if someone is using Leiningen and do not have `clojure` they will still be running with latest jack-in dependencies.